Kudu.Core/Infrastructure/FileSystemHelpers.cs

```diff
@@ -290,6 +289,27 @@ public static void DeleteDirectoryContentsSafe(string path, bool ignoreErrors =
             DeleteDirectoryContentsSafe(Instance.DirectoryInfo.FromDirectoryName(path), ignoreErrors);
         }
 
+        public static bool IsFileSystemReadOnly(IEnvironment environment)
+        {
+            try
+            {
+                // Temp folder will be clean up everytime site starts up
+                // So it is fine to have leftover folder if deletion failed
+                string folderName = Path.Combine(environment.DataPath, "Temp", Guid.NewGuid().ToString("N"));
+                Instance.Directory.CreateDirectory(folderName);
+                Instance.Directory.Delete(folderName);
+                return false;
```

Comment:
Assuming site must restart for readonly/writable state to change, we can cache this state so we avoid too many new i/o operations, which could have side effects. We probably take the lock quite often.